ng-dev/pr/common/fetch-pull-request.ts

```diff
@@ -65,13 +101,92 @@ const PR_SCHEMA = {
   ),
 };
 
-/** A pull request retrieved from github via the graphql API. */
-export type RawPullRequest = typeof PR_SCHEMA;
+export type PullRequestFromGithub = typeof PR_SCHEMA;
 
 /** Fetches a pull request from Github. Returns null if an error occurred. */
 export async function fetchPullRequestFromGithub(
   git: AuthenticatedGitClient,
   prNumber: number,
-): Promise<RawPullRequest | null> {
+): Promise<PullRequestFromGithub | null> {
   return await getPr(PR_SCHEMA, prNumber, git);
 }
+
+/** Fetches a pull request from Github. Returns null if an error occurred. */
+export async function fetchPendingPullRequestsFromGithub(
+  git: AuthenticatedGitClient,
+): Promise<PullRequestFromGithub[] | null> {
+  return await getPendingPrs(PR_SCHEMA, git);
+}
+
+/**
+ * Normalize the statuses for a commit from a pull requeste, using a consistent interface for both
+ * status and checks results.
+ */
+export function getStatusesForPullRequest(pullRequest: PullRequestFromGithub) {
+  /** The combined github status and github checks object. */
+  const {statusCheckRollup} = pullRequest.commits.nodes.slice(-1)[0].commit;
```

Comment:
Nit: I think using the index is more clean and avoids us from cloning the array. e.g.

```ts
  const nodes = pullRequest.commits.nodes;
  const {statusCheckRollup} = nodes[nodes.length - 1].commit;
```